src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -376,6 +376,10 @@ private void OnBatchScopeDisposed()
                             return s.RemoveAdditionalDocument(id);
                         });
 
+                    // Update project's order of documents.
+                    solution = _sourceFiles.UpdateProjectDocumentsOrderForBatch(solution, Id);
```

Comment:
Why not just fold this into UpdateSolutionForBatch?

Reply:
I see what you mean. Yes it should go there.